### PR TITLE
Add the Unit Overview page to Teacher Dashboard

### PR DESCRIPTION
**Pull request by lfryemason**
First step towards making the unit page functional in the teacher dashboard. This PR:
* Adds a unit_summary endpoint that returns all necessary information for displaying the unit overview page
* Adds the unit overview page to the teacher dashboard
* Fixes some printing issues on the teacher dashboard. Includes maintaining printing of existing course and unit overview pages and removing the navigation sidebar from print

image

## Links
https://codedotorg.atlassian.net/browse/TEACH-1263
[Tech Spec](https://docs.google.com/document/d/1o_7Ii02DxzV9D9HvL3-XAs0tJAZUCce2zsJy8AhIQUk/edit)

## Follow-up work

* Add unit_name to URL
* Redirect course option in sidebar to correct URL (with unit or course name)
* Remove teacher panel
* Fix search parameters
* Unit tests that BE endpoint is called and loading state is shown during loading
* [Update selected section](https://github.com/code-dot-org/code-dot-org/pull/61334/files#diff-1851ee33f874e2f56f7535aa3c98c5bb46162a6a7f9c73ce8ee79eb4615da017R135)
* Ensure announcements are working properly

### CODE REVIEW

File: apps/style/curriculum/courses.scss [hunk 1]
```diff
@@ -1,6 +1,7 @@
 @import "markdown";
 @import "pdf_printing";
 
+/* Copied in style/teacherDashboard/teacherDashboard.scss. Update there if anything is changed here. */
```
Comment (Nokondi) [inline apps/style/curriculum/courses.scss]: Just curious - is there a way we could do this without having to manually update these separate files to maintain parity? Maybe importing the CSS from one to the other?

**Review by Nokondi — COMMENTED** [review]
Just have one question!

File: apps/style/curriculum/courses.scss (hunk 1 above)
Comment (lfryemason) [reply apps/style/curriculum/courses.scss]: Great suggestion. I was able to just import the two existing files

**Review by Nokondi — APPROVED** [review]
Looks good!

**Pull request merged**